pkg/failurehandler/apps.go

```diff
@@ -0,0 +1,118 @@
+package failurehandler
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/giantswarm/clustertest"
+	"github.com/giantswarm/clustertest/pkg/application"
+	"github.com/giantswarm/clustertest/pkg/logger"
+
+	appsv1 "k8s.io/api/apps/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	ctrl "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// AppIssues produces debugging information from app-operator (on the MC) and chart-operator (on the WC)
+// This function will log out the status of the deployments, any related Events found and the last 25 lines of logs from the pods.
+func AppIssues(ctx context.Context, framework *clustertest.Framework, cluster *application.Cluster) FailureHandler {
+	return Wrap(func() {
+		logger.Log("Attempting to get debug info for App related failure")
+
+		maxLines := int64(25)
+
+		{
+			appOperatorName := fmt.Sprintf("%s-app-operator", cluster.Name)
```

Comment:
I'm not sure if this logic will work with bundles. iirc the app-operator for bundles is the one in the `giantswarm` namespace.

I'm not sure if you want to add support from the get go to bundles, but at least I'd do a check to skip quickly all further checks if the app is indeed a bundle.
If the app CR has the label `app-operator.giantswarm.io/version: 0.0.0`; we know for sure we don't need to continue.

Reply:
This isn't intended to be used per App. Instead it would be used if the whole app-related test fails (so tries to get some general debugging info)

I hadn't thought about bundles to be honest. How often have we seen only bundles failing to install? I'm not sure we have that I can recall.